src/auth/newuserwindow.cpp

```diff
@@ -0,0 +1,103 @@
+    loginLayout->addWidget(mpLabelLogin);
+    loginLayout->addWidget(mpEditLogin);
+    mainLayout->addLayout(loginLayout);
+
+    //  token/password parameters
+    QHBoxLayout *tokenLayout = new QHBoxLayout;
+    mpLabelToken = new QLabel("Password or token", this);
+    mpLabelToken->setMinimumWidth(mLabelWidth);
+    mpLabelToken->setMaximumWidth(mLabelWidth);
+    mpEditToken = new QLineEdit(this);
+    mpEditToken->setMinimumWidth(mEditWidth);
+    mpEditToken->setMaximumWidth(mEditWidth);
+    mpEditToken->setEchoMode(QLineEdit::Password);
+    mpEditToken->setPlaceholderText(mPlaceholderText);
+    mpEditToken->setToolTip(mPlaceholderText);
+    tokenLayout->addWidget(mpLabelToken);
+    tokenLayout->addWidget(mpEditToken);
+    mainLayout->addLayout(tokenLayout);
+
+    // DialogButtonBox  parameters
+    mpButtonBox = new QDialogButtonBox(QDialogButtonBox::Ok | QDialogButtonBox::Cancel, this);
+    //mpButtonBox->button( QDialogButtonBox::Ok )->setEnabled( false );
+    connect(mpButtonBox, &QDialogButtonBox::clicked, this, &NewUserWindow::onBtnBoxClicked);
+    mainLayout->addWidget(mpButtonBox);
+
+    // separator
+    QFrame *line = new QFrame(this);
+    line->setFrameShape(QFrame::HLine);
+    line->setFrameShadow(QFrame::Sunken);
+    line->setLineWidth(mSeparatorWidth);
+    mainLayout->addWidget(line);
+
+    // unnamed user button
+    QHBoxLayout *unnamedButtonLayout = new QHBoxLayout;
+    mpUnnamedUserButton = new QPushButton("Try without authorizattion", this);
+    mpUnnamedUserButton->setMinimumWidth(mUnnamedUserButtonWidth);
+    mpUnnamedUserButton->setMaximumWidth(mUnnamedUserButtonWidth);
+    unnamedButtonLayout->addWidget(mpUnnamedUserButton);
+    mainLayout->addLayout(unnamedButtonLayout);
+    connect(mpUnnamedUserButton, &QPushButton::clicked, this, [=](){emit unnamedUser();});
+    setLayout(mainLayout);
+    show();
+    move((QApplication::desktop()->width() - this->width()) / 2 ,
+         (QApplication::desktop()->height() - this->height()) / 2);
+}
+
+void NewUserWindow::onBtnBoxClicked(QAbstractButton *button)
+{
+    QDialogButtonBox::StandardButton pressedButton;
+    pressedButton = mpButtonBox->standardButton(button);
+
+    if (pressedButton == QDialogButtonBox::Cancel)
+    {
+        emit cancel();
+        close();
+    }
+
+    if (pressedButton == QDialogButtonBox::Ok)
+    {
+        if (!mpEditToken->text().size() || !mpEditLogin->text().size())    // login or token not typed
+        {
+            return;
+        }
+        if (mpEditToken->text().size() == 40 && mpEditLogin->text().size())// user typed token
+        {
+            emit newUserToken(mpEditLogin->text(), mpEditToken->text());
+            return;
+        }
+        if (mpEditToken->text().size() && mpEditLogin->text().size())      // user typed password
+        {
+            emit newUserPasssword(mpEditLogin->text(), mpEditToken->text());
+            return;
+        }
+    }
```

Comment:
1) why you just use else if/else? this fucntion doesn't need to use so much returns enywhere.
2) you use the simular condition in every "if" statement. That's not good because this funtion will do the same checks averytime.
3) As I understood, there are only 3 ways:

**1. User typed token
2. User typed password
3. All that remained / user hasn't typed at least one of the field.**

Third confiditon is default, which will happen if 1-st and 2-nd are false.

Also In 1-st and 2-nd statment you use the same condition, avoid it

You can rewrite this part of code as

```
if(mpEditLogin->text().size())
{
    if (mpEditToken->text().size() == 40 
    {
        emit newUserToken(mpEditLogin->text(), mpEditToken->text());
    }
    else
    {
        emit newUserPasssword(mpEditLogin->text(), mpEditToken->text());
    }
}
```
here you don't need returns and you calculate lesser count of operations